From a856c86188c30d493decc3351f757f33e75f5435 Mon Sep 17 00:00:00 2001
From: "Victor M. Alvarez" <vmalvarez@virustotal.com>
Date: Fri, 17 Dec 2021 12:50:48 +0100
Subject: [PATCH 2/2] Use YR_RE_SCAN_LIMIT as limit instead of MAX_INT.

---
 libyara/re.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/libyara/re.c b/libyara/re.c
index 6ddd1910..604b463c 100644
--- a/libyara/re.c
+++ b/libyara/re.c
@@ -2085,10 +2085,13 @@ int yr_re_fast_exec(
   RE_FAST_EXEC_POSITION* last;
 
   int input_incr = flags & RE_FLAGS_BACKWARDS ? -1 : 1;
-  int max_bytes_matched = flags & RE_FLAGS_BACKWARDS
-                              ? (int) yr_min(input_backwards_size, INT_MAX)
-                              : (int) yr_min(input_forwards_size, INT_MAX);
   int bytes_matched;
+  int max_bytes_matched;
+
+  if (flags & RE_FLAGS_BACKWARDS)
+    max_bytes_matched = (int) yr_min(input_backwards_size, YR_RE_SCAN_LIMIT);
+  else
+    max_bytes_matched = (int) yr_min(input_forwards_size, YR_RE_SCAN_LIMIT);
 
   const uint8_t* ip = code;
 
-- 
2.34.0

